include/fuse.h

```diff
@@ -113,9 +113,13 @@ struct fuse_config {
 	/**
 	 * If `set_mode` is non-zero, the any permissions bits set in
 	 * `umask` are unset in the st_mode attribute of each file.
+	 * `fmask` and `dmask` may be used instead of `umask` to affect
+	 * files and directories separately, respectively.
 	 */
 	int set_mode;
 	unsigned int umask;
+	unsigned int fmask;
+	unsigned int dmask;
```

Comment:
I think this is going to break the ABI (we currently have no test for that, but I'm working on that). Please see this PR https://github.com/libfuse/libfuse/pull/978

Reply:
i interpreted "The remaining options are used by libfuse internally and should not be touched" to imply "not part of the abi" and accordingly placed fmask and dmask before those fields as opposed to at the literal end of the structure; is that correct? or shall i move them to the literal end?

Reply:
Yeah it is confusing, I'm changing that in my PR. And also asked for review from other maintainers for PR#978.
From my point of view your additions have to go below as otherwise values are shifted by 64 bit.